plugins/woocommerce/src/StoreApi/Schemas/V1/CheckoutSchema.php

```diff
@@ -339,7 +339,12 @@ function( $carry, $key ) use ( $fields, $properties ) {
 				}
 				$field_schema   = $properties[ $key ];
 				$rest_sanitized = rest_sanitize_value_from_schema( wp_unslash( $fields[ $key ] ), $field_schema, $key );
-				$carry[ $key ]  = wp_kses( $rest_sanitized, [] );
+				$carry[ $key ]  = $rest_sanitized;
+
+				// Specific sanitization for string types, skipping other types from being coerced to strings.
+				if ( 'string' === $field_schema['type'] ) {
+					$carry[ $key ] = wp_kses( $rest_sanitized, [] );
+				}
```

Comment:
Should this be done in the address as well?

Reply:
Added in c283b95df2ed0038bd4e76cfddbb42f02e46a4b0